### ISSUE TITLE
Wrapped define calls (e.g. UMD) not resolved during build

### ISSUE THREAD

**Issue by webpro** [jrburke/r.js#414]
**tl;dr** Wrapped 'define' call to named module (e.g. UMD) as build dependency is not resolved during build.

Scenario:
- A built file (optimized by r.js), say core.js.
- This core.js has a UMD-styled module included, say myLib.js.
- The core.js as a whole is a dependency in a build config of some other JS app (in `deps:[]` config), say app.js
- During the build of app.js, "myLib" is a named module in core.js, but it is not resolved as a define'd module (because it's wrapped / UMD boilerplate).
- Now r.js is trying to load "myLib" locally from fs where it can't find it.

The issue, I think, is the wrapping of the `define` call in a function, which doesn't get resolved.

For example, "myLib" ends up in a build (core.js) like this:

```
define("myLib", [], function() {
    return"myLib"
})
```

But with UMD boilerplate for example it's like this:

```
function(e, t) {
    typeof exports == "object" ? module.exports = t() : typeof define == "function" && define.amd ? define("myLib", t) : e.myLib = t()
}(this, function() {
    return"myLib"
})
```

Or this:

```
function(e) {
    e("myLib", [], function() {
        return"myLib"
    })
}(typeof define == "function" && define.amd ? define : function(e) {
    module.exports = e()
})
```

For app.js, "myLib" is not caught as a dependency. Any thoughts on this?

I'm not in control of how 3rd party libs do this UMD or global-or-define logic.

I could include "myLib" in the app, and exclude it from the build; or use the "empty!" plugin trick or something. But I'd really rather not require a work-around in the app, but somehow fix it in the core project, or its build process instead.
